File: 双指针/简单/回文链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function (head) {
  var stack = []
  while (head) {
    stack.push(head.val)
    head = head.next
  }
  var l = 0,
    r = stack.length - 1
  while (l < r) {
    if (stack[l] !== stack[r]) {
      return false
    }
    l++
    r--
  }
  return true
}

File: 链表/简单/回文链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        ListNode* p=head;
        int n=0;
        while(p){
            n++;
            p=p->next;
        }
        stack<int> s1;
        int middle = n/2;
        while(middle--){
            s1.push(head->val);
            head=head->next;
        }
        if(n%2){
            head=head->next;
        }
        while(s1.size()){
            if(s1.top()!=head->val){
                return false;
            }
            s1.pop();
            head=head->next;
        }
        return true;
    }
};